src/gluonts/representation/global_relative_binning.py

```diff
@@ -0,0 +1,170 @@
+        self,
+        num_bins: int = 1024,
+        is_quantile: bool = True,
+        linear_scaling_limit: int = 10,
+        quantile_scaling_limit: float = 0.99,
+        *args,
+        **kwargs,
+    ):
+        super().__init__(*args, **kwargs)
+
+        self.num_bins = num_bins
+        self.is_quantile = is_quantile
+
+        self.linear_scaling_limit = linear_scaling_limit
+        self.quantile_scaling_limit = quantile_scaling_limit
+
+        self.bin_edges = self.params.get_constant(
+            "bin_edges", mx.nd.zeros(self.num_bins + 1)
+        )
+        self.bin_centers = self.params.get_constant(
+            "bin_centers", mx.nd.zeros(self.num_bins)
+        )
+
+    def initialize_from_dataset(
+        self, input_dataset: Dataset, ctx: mx.Context = get_mxnet_context()
+    ):
+        # Rescale all time series in training set.
+        train_target_sequence = np.array([])
+        for train_entry in input_dataset:
+            train_entry_target = train_entry["target"]
+            train_tar_mean = np.mean(train_entry_target)
+            train_entry_target /= train_tar_mean
+            train_target_sequence = np.concatenate(
+                [train_target_sequence, train_entry_target]
+            )
+        self.initialize_from_array(train_target_sequence, ctx)
+
+    def initialize_from_array(
+        self, input_array: np.ndarray, ctx: mx.Context = get_mxnet_context()
+    ):
+        # Calculate bin centers and bin edges using linear or quantile binning..
+        if self.is_quantile:
+            bin_centers = np.quantile(
+                input_array,
+                np.linspace(0, self.quantile_scaling_limit, self.num_bins),
+            )
+            bin_centers = ensure_binning_monotonicity(bin_centers)
+        else:
+            has_negative_data = np.any(input_array < 0)
+            low = -self.linear_scaling_limit if has_negative_data else 0
+            high = self.linear_scaling_limit
+            bin_centers = np.linspace(low, high, self.num_bins)
+        bin_edges = bin_edges_from_bin_centers(bin_centers)
+
+        # Store bin centers and edges since their are globally applicable to all time series.
+        with ctx:
+            self.bin_edges.initialize()
+            self.bin_centers.initialize()
+        self.bin_edges.set_data(mx.nd.array(bin_edges))
+        self.bin_centers.set_data(mx.nd.array(bin_centers))
+
+    # noinspection PyMethodOverriding
+    def hybrid_forward(
+        self,
+        F,
+        data: Tensor,
+        observed_indicator: Tensor,
+        scale: Optional[Tensor],
+        rep_params: List[Tensor],
+        **kwargs,
+    ) -> Tuple[Tensor, Tensor, List[Tensor]]:
+        bin_edges = kwargs["bin_edges"]
+        bin_centers = kwargs["bin_centers"]
+
+        # Calculate local scale if scale is not already supplied.
+        if scale is None:
+            scale = F.expand_dims(
+                F.sum(data, axis=-1) / F.sum(observed_indicator, axis=-1), -1
```

Comment:
Looks like the scale here is computed as the average observation: however the numerator assumes that unobserved data is simply zero, which may not necessarily be the case. Maybe the numerator could be
```
F.sum(data * observed_indicator, axis=-1)
```
The same holds anywhere else the scale is computed this way

Reply:
Related to this: is it necessary to handle the scaling in the binning components? If I understand correctly the design, binnings use `rep_params` to pass down the information to `post_transform` about how to reconstruct the data in the original domain: couldn't scalers do the same? This way `scale` would disappear from `hybrid_forward` and `post_transform`, and just be included in `rep_params` by the scaler, and the "scaling+binning" could be implemented as a chain of the two.

But that I guess would require some strategy on how to organize the `rep_params` in composed representations: for example, a chain could use some sort of stack.

Do I understand correctly that right now it is assumed that a chain contains at most one binning and that any other representation should not "store" anything in its output `rep_params`?